### ISSUE TITLE
Chào bạn

### ISSUE THREAD

**Issue by thanhlouis** [duykhongphai/Dragonboy-Captcha-Ocr#1]
Này chỉ dành riêng cho captcha của game này thôi hay có thể sử dụng train cho captcha khác được vậy bạn. Cảm ơn

**Comment by duykhongphai** [comment]
riêng cho game này thôi b

**Comment by thanhlouis** [comment]
Cho mình hỏi là thử xài google colab của bài này: https://keras.io/examples/vision/captcha_ocr/
xong dùng model.save('captcha_ocr_model.h5') để lấy model rồi về tìm cách chạy cho nó giải mà loay hoay mãi không biết cách cho nó chạy <(") Bạn có thể cho mình code mẫu được không. Cảm ơn

**Comment by duykhongphai** [comment]
from tensorflow import keras
model = keras.models.load_model('aptcha_ocr_model.h5l')

characters = [... như trong lúc traning]

char_to_num = keras.layers.StringLookup(
    vocabulary=characters, mask_token=None
)

num_to_char = keras.layers.StringLookup(
    vocabulary=char_to_num.get_vocabulary(), mask_token=None, invert=True
)

img_width = 200
img_height = 50

def preprocess_image(image_path):
    # Đọc ảnh và chuyển sang grayscale
    img = keras.preprocessing.image.load_img(
        image_path, color_mode="grayscale", target_size=(img_height, img_width)
    )
    # Chuyển sang mảng numpy và chuẩn hóa
    img = keras.preprocessing.image.img_to_array(img)
    img = img.astype("float32") / 255.0
    # Mở rộng chiều để phù hợp với đầu vào của model
    img = np.expand_dims(img, axis=0)
    return img

def decode_predictions(pred):
    input_len = np.ones(pred.shape[0]) * pred.shape[1]
    results = keras.backend.ctc_decode(pred, input_length=input_len, greedy=True)[0][0]
    # Chuyển đổi indices sang characters
    output_text = []
    for res in results:
        res = tf.gather(num_to_char.get_vocabulary(), res)
        output_text.append(tf.strings.reduce_join(res).numpy().decode("utf-8"))
    return output_text

image_path = 'captcha.png'
processed_img = preprocess_image(image_path)
prediction = model.predict(processed_img)
predicted_text = decode_predictions(prediction)
print("Captcha được nhận diện là:", predicted_text[0])

Code tham khảo nha
